src/Contracts/UserRepository.php

```diff
@@ -58,4 +58,28 @@ public function createUser(String $email, String $firstName, String $lastName):
      * @param mixed $id
      */
     public function delete($id);
+
+    /**
+     * Get all assigned roles of the given user.
+     *
+     * @param string $userId
+     * @return Collection
+     */
+    public function getRoles(string $userId): Collection;
+
+    /**
+     * Assign the given roles to the given user.
+     *
+     * @param string $userId
+     * @param Collection $roleIds
+     */
+    public function addRoles(string $userId, Collection $roleIds): void;
+
+    /**
+     * Remove the given roles from the given user's assigned roles.
```

Comment:
Does this mean that an intersection of the given roles and the user's roles is made, and those removed? Or will it error if not all given roles are assigned to the user? I know this check is delegated to Auth0, but what actually happens?

Reply:
It will just make sure the given roles won't be assigned to the user anymore after returning. If the user didn't have some/all of the given roles to begin with, those roles are simply ignored.

Reply:
I would then add a small extra line saying that roles that the user doesn't have will be ignored, just to avoid confusion.

Reply:
Updated the php docs.